### ISSUE TITLE
Binding Examples [Discussion]

### ISSUE THREAD

**Issue by birm** [mlpack/examples#97]
I'm not confident if this is a good idea, but it would be interesting to have a simple example for each of the bindings. This could double as a more robust binding test, as well as a demonstration. It may, however, add unneeded complexity.
I'm not sure. Thoughts?

**Comment by kartikdutt18** [comment]
I think it's a nice idea. Since the examples repo is not restricted to models, It would be a good idea to showcase various features of mlpack including the bindings.

**Comment by rcurtin** [comment]
I agree, it would be really nice.  For now we just have the quickstart examples, but perhaps those could be additionally moved here and then expanded on.  Some of the (very old) mlpack tutorials also have binding usage examples, but they aren't always great examples.

**Comment by abernauer** [comment]
Yeah when writing that example with R binding I tried to expand on the quick start example and fit it into a standard machine learning workflow. I think for newcomers to mlpack having an example with how to use a binding in data science workflow is helpful. I think it's a solid idea.

**Comment by rcurtin** [comment]
@birm do you think there is anything else to do for this issue? It seems like everyone is agreed that binding examples would definitely be a good idea. :)

**Comment by birm** [comment]
Should I edit this issue to include subtasks for each binding then?

**Comment by rcurtin** [comment]
@birm up to you; if you want to keep it open until we have examples of each binding, that sounds good to me.  Or, it might be easier to open a new issue for it and tag it with `help-wanted` `good-first-issue`, so that newcomers can quickly get started and see what needs to be done, as opposed to scrolling through the discussion here.  Your call though. :)
